src/info/repo.rs

```diff
@@ -191,14 +188,11 @@ impl Repo {
         Ok(number_of_branches)
     }
 
-    pub fn get_git_username(&self) -> Result<String> {
-        let config = self.git2_repo.config()?;
-        let username = match config.get_entry("user.name") {
-            Ok(v) => v.value().unwrap_or("").into(),
-            Err(_) => "".into(),
-        };
-
-        Ok(username)
+    pub fn get_git_username(&self) -> String {
+        self.repo
+            .committer()
+            .map(|c| c.name.to_string())
+            .unwrap_or_default()
```

Comment:
What do you think about defaulting to something like `"[UNKNOWN]"` if we cannot unwrap the committer? Technically a `user.name` *could* be `[UNKNOWN]`, but I think it's unlikely enough 😆 

Or if we stick with the default, why not use [`committer_or_default`](https://docs.rs/git-repository/0.20.0/git_repository/struct.Repository.html#method.committer_or_default)?

Reply:
A great catch, for this round I left it functionally similar, so if nothing is configured there are empty strings. This behaviour is also present in `repo_name_and_url()` even though `null` would probably bode better in with json output, I don't know if the UI has any special handling for these empty strings though. Generally I'd certainly prefer options.

This also answers why `committer_or_default()` wasn't chosen - you actually get an entirely made up user name `"gitoxide <gitoxide@localhost>"` which probably is misleading - it's akin to what git does if nothing is configured, using the current user name along with some way of figuring out a local email address.

To sum it up, I'd prefer returning `Option` instead of empty strings and have the UI decide how to display these.

Reply:
>for this round I left it functionally similar

Got it, that's probably for the best :+1:
Sorry, I should have checked the original behavior more closely